common/api/appui-react.api.md

```diff
@@ -3328,6 +3334,24 @@ export interface PanelStateChangedEventArgs {
     panelState: StagePanelState;
 }
 
+// @alpha (undocumented)
+export type PanelsToolbarItem = ToolbarItem & {
+    readonly usage: ToolbarUsage;
+    readonly orientation: ToolbarOrientation;
+};
+
+// @alpha (undocumented)
+export interface PanelsUiItemsProvider extends UiItemsProvider {
+    readonly getToolbarItems?: () => ReadonlyArray<PanelsToolbarItem>;
+    readonly getWidgets?: () => ReadonlyArray<PanelsWidget>;
+}
+
+// @alpha (undocumented)
+export interface PanelsWidget extends Widget {
```

Comment:
@jason-crow, @raplemie does this make more sense? How about naming?

Reply:
I think its a good direction, that will certainly allow us to have different layout in the future, and have more extensible options to them, I'm only wondering if we won't risk have clashes in interpreting these without a clear discriminator, or how would someone provide information to multiple layouts simultaneously if ever 2 have clashing names with different semantics.
Could something like this be a good idea ?
```ts
interface Widget {
  ...(current existing widget props),
  layoutConfig: {[layoutId: string]: {} },
}

const myWidget: Widget = {
  id: "my-widget",
  ...props,
  layoutConfig: {
    panels: {
      location: StagePanelLocation.Top,
      section:...,
    },
    spatial: {
      location: SpatialLocation.DocumentPanel,
    }
  }
}
```

Or we dont expect the provider creator to provide default values, so there wouldnt be any clashes ?

Reply:
Applied this proposal to both `Widget` and `ToolbarItem` types. How do we want to name the `panels` layout?

Reply:
Updated `UiItemsManager` to return widgets/toolbar items w/o a defined location (for consistency with other item types).